internal/services/pam/pam_test.go

```diff
@@ -0,0 +1,623 @@
+			t.Cleanup(cleanup)
+
+			abResp, err := client.AvailableBrokers(context.Background(), &authd.Empty{})
+			if tc.wantErr {
+				require.Error(t, err, "AvailableBrokers should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "AvailableBrokers should not return an error, but did")
+			got := abResp.GetBrokersInfos()
+			want := testutils.LoadWithUpdateFromGoldenYAML(t, got)
+			require.Equal(t, want, got, "AvailableBrokers returned unexpected brokers")
+		})
+	}
+}
+
+func TestSetDefaultBrokerForUser(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		username string
+		noBroker bool
+
+		wantErr bool
+	}{
+		"Set default broker for existing user": {username: "user"},
+
+		"Error when username is empty":     {wantErr: true},
+		"Error when broker does not exist": {username: "user", noBroker: true, wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			client, cleanup := startClient(t, filepath.Join(fixturesPath, "mixed_brokers"))
+			t.Cleanup(cleanup)
+
+			abResp, err := client.AvailableBrokers(context.Background(), &authd.Empty{})
+			require.NoError(t, err, "Setup: failed to get available brokers for tests")
+
+			//nolint:gosec // This just to vary the selected broker in the tests, we don't care about randomness safety.
+			wantID := abResp.GetBrokersInfos()[rand.Intn(len(abResp.GetBrokersInfos()))].GetId()
+			if tc.noBroker {
+				wantID = "does not exist"
+			}
+
+			sdbfuResquest := &authd.SDBFURequest{
+				BrokerId: wantID,
+				Username: tc.username,
+			}
+			_, err = client.SetDefaultBrokerForUser(context.Background(), sdbfuResquest)
+			if tc.wantErr {
+				require.Error(t, err, "SetDefaultBrokerForUser should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "SetDefaultBrokerForUser should not return an error, but did")
+
+			gotResp, _ := client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: tc.username})
+			require.Equal(t, wantID, gotResp.GetPreviousBroker(), "SetDefaultBrokerForUser did not set the correct broker for the user")
+		})
+	}
+}
+
+func TestGetPreviousBroker(t *testing.T) {
+	t.Parallel()
+
+	client, cleanup := startClient(t, filepath.Join(fixturesPath, "mixed_brokers"))
+	t.Cleanup(cleanup)
+
+	_, err := client.SetDefaultBrokerForUser(context.Background(), &authd.SDBFURequest{
+		BrokerId: "local",
+		Username: "user",
+	})
+	require.NoError(t, err, "Setup: could not set default broker for user for tests")
+
+	gotResp, _ := client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: "user"})
+	require.Equal(t, "local", gotResp.GetPreviousBroker(), "GetPreviousBroker did not return the correct broker")
+
+	gotResp, _ = client.GetPreviousBroker(context.Background(), &authd.GPBRequest{Username: "does not have a broker"})
+	require.Empty(t, gotResp.GetPreviousBroker(), "GetPreviousBroker should return nil when the user has no broker assigned")
```

Comment:
GetPreviousBroker does not return an error ever, so it would be a pointless assertion. Maybe we should update the API?

```Go
func (s Service) GetPreviousBroker(ctx context.Context, req *authd.GPBRequest) (*authd.GPBResponse, error) {
	var r authd.GPBResponse
	b := s.brokerManager.BrokerForUser(req.GetUsername())
	if b != nil {
		r.PreviousBroker = &b.ID
	}
	return &r, nil
}
```